xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -799,6 +804,43 @@ HRESULT CVideoBufferShared::GetResource(ID3D11Resource** ppResource)
   if (SUCCEEDED(hr))
     hr = m_sharedRes.CopyTo(ppResource);
 
+  // Failures in the synchronization code are not blocking problems, log but return success
+  if (!m_sharedFence && m_handleFence != INVALID_HANDLE_VALUE)
+  {
+    HRESULT hr2;
+    // open the fence on app device
+    ComPtr<ID3D11Device> pD3DDevice = DX::DeviceResources::Get()->GetD3DDevice();
+    ComPtr<ID3D11Device5> device5;
+    if (FAILED(hr2 = pD3DDevice.As(&device5)))
+    {
+      CLog::LogF(LOGDEBUG, "ID3D11Device5 is not available, error description: {}",
+                 DX::GetErrorDescription(hr2));
+    }
+    else if (FAILED(hr2 = device5->OpenSharedFence(m_handleFence, IID_PPV_ARGS(&m_sharedFence))))
+    {
+      CLog::LogF(LOGDEBUG, "unable to open the shared fence, error description: {}",
+                 DX::GetErrorDescription(hr2));
+    }
+  }
+
+  if (m_sharedFence)
+  {
+    // Make the GPU wait for the fence value that produced the picture
+    ComPtr<ID3D11DeviceContext1> context1 = DX::DeviceResources::Get()->GetImmediateContext();
+    ComPtr<ID3D11DeviceContext4> context4;
```

Comment:
Is not necessary obtain this interface for every frame during playback. This code block can be moved inside previous if block (executed only at initialization):

` if (!m_sharedFence && m_handleFence != INVALID_HANDLE_VALUE)`

and save `ID3D11DeviceContext4` as member variable.

Then in every frame is enough execute `m_context4->Wait` line